IronFoundry.Warden.Shared/ExtensionMethods.cs

```
@@ -38,6 +39,29 @@ public static Security.SecureString ToSecureString(this string unsecuredString)
             return securedString;
         }
     }
+
+    public static class ActionExtensionMethods
+    {
+        public static bool ThrowsException<T>(this Action action) where T : Exception
+        {
+            bool threwException = false;
+
+            try
+            {
+                action();
+            }
+            catch (T)
+            {
+                threwException = true;
+            }
+            catch
+            {
+                // ignored
```

We shouldn't ignore unexpected exceptions.  Let the exception bubble up.